include/nbl/system/IAsyncQueueDispatcher.h

```diff
@@ -19,10 +19,20 @@ namespace impl
             // wait on this for result to be ready
             std::condition_variable cvar;
             bool ready = false;
```

Comment:
shouldn't this be an atomic boolean or atomic_flag?

Reply:
it could but pointless because it's always touched when a thread is holding queue mutex

Reply:
> it's always touched when a thread is holding queue mutex
I'd want to get rid of this global lock